### ISSUE TITLE
Widget not updating 

### ISSUE THREAD

**Issue by xandro0777** [woheller69/omweather#114]
I have update interval set to 15 minutes but the widget apparently never updates unless the app is opened.

I have this and your Gas Prices widget on the same screen with the same settings, gas prices update as expected, weather not.

F-Droid, Android 11

**Comment by woheller69** [comment]
Duplicate of #95

**Comment by xandro0777** [comment]
So I have disabled battery optimisation for this app, even allowed background activity and nothing helps.

Also, the gas prices widget works fine without any of this tweaks as well as many other widgets.

Any idea?

**Comment by woheller69** [comment]
No, it is essentially the same app

**Comment by xandro0777** [comment]
Somehow there is a huge difference. As described before, Gas Prices is attributed almost 15% Battery usage while Cirrus Weather 0.34%

**Comment by woheller69** [comment]
These attributions are total nonsense ...

**Comment by xandro0777** [comment]
Apparently it works fine after a reboot. Probably the battery optimisation setting change took effect only after reboot 🤔

Some Apps ask the user to change this setting and directly invoke the relevant menu.. would be nice to have.

**Comment by woheller69** [comment]
widgets are managed by the Android system.
If a widget hangs you need to reboot or remove the widget and add it again
